### PR TITLE
feat(kinsta): add interactive site and environment selection for env push

### PR DESCRIPTION
**Pull request by codepuncher**
## Summary

Add interactive site/environment selection to `iroots kinsta env push`, aligned with `env open`, while preserving ID-based automation paths.

## Related

- Branch: `feat/kinsta-env-push-ui`

## Changes

- Added shared Kinsta selector utilities for site/environment matching and disambiguation prompts.
- Refactored `kinsta env open` to use shared selector logic with preserved explicit-flag fail-fast behaviour.
- Extended `kinsta env push` with friendly flags (`--site`, `--source_env`, `--target_env`) and interactive resolution when IDs are not fully supplied.
- Enforced strict and deterministic ID/name conflict handling for push target resolution.
- Added and updated automated tests for selector behaviour and push resolution edge cases.
- Updated project test script and regenerated command docs in `README.md`.

## Testing

- `npm test`
- `npm run lint`
- `npm run build`

### CODE REVIEW

File: src/lib/kinsta-selectors.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+async function promptForSite(sites: KinstaSite[], message: string): Promise<KinstaSite> {
+  const sortedSites = [...sites].sort((a, b) => a.display_name.localeCompare(b.display_name))
+
+  const siteId = await select({
+    message,
+    choices: sortedSites.map(site => ({
+      name: `${site.display_name} (${site.name}) [${site.id}] ${site.environments ? `envs: ${site.environments.map(e => e.display_name || e.name).join(', ')}` : 'envs: unknown'}`,
+      value: site.id,
+    })),
```
[the remark this answers is not in the document]
Comment (codepuncher) [reply src/lib/kinsta-selectors.ts:50]: Fixed in 78fab0a and refined in 82e4e29. Site choice summaries now sort environment names and fall back to `envs: unknown` when environments are missing/empty.

File: src/lib/kinsta-selectors.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+      value: environment.id,
+    })),
+  })
+
+  return sortedEnvironments.find(environment => environment.id === environmentId) ?? sortedEnvironments[0]
+}
+
+export async function resolveSite(sites: KinstaSite[], candidates: string[], explicitSite?: string): Promise<KinstaSite> {
+  const explicit = normalizeOptionalFlag(explicitSite)
+  if (explicit) {
```
[the remark this answers is not in the document]
Comment (codepuncher) [reply src/lib/kinsta-selectors.ts]: Fixed in 78fab0a. Updated `resolveSite()` to use explicit `undefined` checks (`explicit !== undefined`) instead of truthiness.

File: src/lib/kinsta-selectors.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+}
+
+export async function resolveEnvironment(environments: KinstaEnvironment[], candidates: string[], explicitEnvironment?: string): Promise<KinstaEnvironment> {
+  const explicit = normalizeOptionalFlag(explicitEnvironment)
+  if (explicit) {
+    const matches = findMatchingEnvironments(environments, explicit)
+    if (matches.length === 1) return matches[0]
+    if (matches.length > 1) return promptForEnvironment(matches, `Multiple environments matched --environment "${explicitEnvironment}". Select one:`)
+    throw new Error(`No environment matched --environment "${explicitEnvironment}"`)
+  }
```
[the remark this answers is not in the document]
Comment (codepuncher) [reply src/lib/kinsta-selectors.ts]: Fixed in 78fab0a and refined in 82e4e29. `resolveEnvironment()` now accepts configurable flag labels and selection prompts; `env push` now uses explicit FROM/TO wording (`--source_env` and `--target_env`) to avoid ambiguity.

File: src/lib/kinsta-selectors.ts [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+  if (uniqueInferredMatches.length === 1) return uniqueInferredMatches[0]
+  if (uniqueInferredMatches.length > 1) return promptForSite(uniqueInferredMatches, 'Multiple inferred sites matched. Select a Kinsta site:')
+
+  if (sites.length > 0) return promptForSite(sites, 'Select a Kinsta site:')
+  throw new Error('No sites available')
+}
+
+export async function resolveEnvironment(environments: KinstaEnvironment[], candidates: string[], explicitEnvironment?: string): Promise<KinstaEnvironment> {
+  const explicit = normalizeOptionalFlag(explicitEnvironment)
+  if (explicit) {
```
[the remark this answers is not in the document]
Comment (codepuncher) [reply src/lib/kinsta-selectors.ts]: Fixed in 78fab0a. Updated `resolveEnvironment()` to use explicit `undefined` checks for clarity and consistency.

File: src/commands/kinsta/env/push.ts [hunk 5]
```diff
@@ -1,26 +1,149 @@
...
+    : findById(siteId, sites) ?? await resolveSite(sites, compact([siteId, site]), site)
+
+  if (siteId !== undefined && site !== undefined) {
+    const matchingSites = findMatchingSites(sites, site)
+    if (!matchingSites.some((matchingSite) => toLower(matchingSite.id) === toLower(selectedSite.id))) {
+      throw new Error(`--site_id "${siteId}" does not match --site "${site}".`)
+    }
+  }
+
+  const environments = selectedSite.environments ?? await input.getSiteEnvironments(input.apiKey, selectedSite.id)
```
[the remark this answers is not in the document]
Comment (codepuncher) [reply src/commands/kinsta/env/push.ts]: Fixed in 89535fe. `env push` now fetches environments when the preloaded list is missing **or empty**, matching the `env open` behaviour and avoiding false `No environments available` failures.

File: README.md [hunk 6]
```diff
@@ -339,118 +952,169 @@ Create a new project
...
+                                                send requests to their API
+      --[no-]sentry                             whether or not to create a Sentry project
+      --sentry_api_key=<value>                  (required) [env: IROOTS_SENTRY_API_KEY] The API key
+      --sentry_organisation_slug=<value>        (required) [env: IROOTS_SENTRY_ORGANISATION_SLUG] The slug of the
+                                                organization the resource belongs to.
+      --sentry_project_default_rules            Defaults to true where the behavior is to alert the user on every new
+                                                issue. Setting this to false will turn this off and the user must create
+                                                their own alerts to be notified of new issues.
+      --sentry_project_platform=<value>         (required) [default: php] The platform for the project.
+      --sentry_project_slug=<value>             Uniquely identifies a project. If ommitted, we will use the project
```
[the remark this answers is not in the document]
Comment (codepuncher) [reply README.md]: Fixed in 89535fe. Corrected the generated docs typo from `ommitted` to `omitted`.

File: README.md [hunk 7]
```diff
@@ -872,12 +1878,13 @@ FLAGS
   --follow_redirects=<value>
   --host=<value>
   --include_header=<value>
-  --name=<value>              (required) Name of the check
+  --name=<value>              Name of the check. If ommitted, we will extract the domain from --website_url and use
```
[the remark this answers is not in the document]
Comment (codepuncher) [reply README.md]: Fixed in 89535fe. Corrected this generated docs typo from `ommitted` to `omitted` as well.

File: src/commands/kinsta/env/push.ts [hunk 8]
```diff
@@ -1,26 +1,152 @@
...
+    const matchingSites = findMatchingSites(sites, site)
+    if (!matchingSites.some((matchingSite) => toLower(matchingSite.id) === toLower(selectedSite.id))) {
+      throw new Error(`--site_id "${siteId}" does not match --site "${site}".`)
+    }
+  }
+
+  const preloadedEnvironments = selectedSite.environments ?? []
+  const environments = preloadedEnvironments.length > 0
+    ? preloadedEnvironments
+    : await input.getSiteEnvironments(input.apiKey, selectedSite.id)
```
[the remark this answers is not in the document]
Comment (codepuncher) [reply src/commands/kinsta/env/push.ts]: Fixed in 011fe91. `resolvePushTargetIds()` now supports partial-ID resolution with `--site_id` without requiring `--company`, by resolving environments directly via `getSiteEnvironments()`. Added regression coverage to ensure `getAllSites()` is not called in that path.

File: src/commands/kinsta/env/push.ts [hunk 9]
```diff
@@ -62,13 +198,47 @@ export default class Push extends KinstaCommand {
...
+        targetEnvId: targetEnvIdFlag,
+      })
+    } catch (error: unknown) {
+      const message = error instanceof Error ? error.message : String(error)
+      this.error(message)
+    }
+
+    const environments = await getSiteEnvironments(flags.apiKey, resolvedIds.siteId)
+    const sourceEnvName = environments.find((environment) => environment.id === resolvedIds.sourceEnvId)?.display_name ?? resolvedIds.sourceEnvId
+    const targetEnvName = environments.find((environment) => environment.id === resolvedIds.targetEnvId)?.display_name ?? resolvedIds.targetEnvId
```
[the remark this answers is not in the document]
Comment (codepuncher) [reply src/commands/kinsta/env/push.ts]: Fixed in 9287fef. `resolvePushTargetIds()` now optionally returns resolved environment display names for partial/interactive paths, and `run()` reuses those names instead of unconditionally fetching environments again. `run()` now only fetches environments for naming in the full-ID path where names were not previously resolved.

File: src/commands/kinsta/env/push.ts [hunk 10]
```diff
@@ -1,26 +1,179 @@
...
+  const targetEnv = normalizeOptionalFlag(input.targetEnv)
+
+  const hasAllIds = siteId !== undefined && sourceEnvId !== undefined && targetEnvId !== undefined
+  if (hasAllIds) {
+    if (sourceEnvId === targetEnvId) {
+      throw new Error('Source and target environments must be different.')
+    }
+
+    return {siteId, sourceEnvId, targetEnvId}
+  }
```
[the remark this answers is not in the document]
Comment (codepuncher) [reply src/commands/kinsta/env/push.ts]: Good catch — fixed in 9ca109c. The full-ID fast path now compares source and target environment IDs case-insensitively, so values like `env-a` and `ENV-A` are rejected as the same environment.

File: src/commands/kinsta/env/push.ts [hunk 11]
```diff
@@ -1,26 +1,254 @@
...
+
+const resolveSiteAndEnvironments = async (
+  input: ResolvePushTargetIdsInput,
+  siteId: string | undefined,
+  site: string | undefined,
+): Promise<{environments: KinstaEnvironments; selectedSiteId: string}> => {
+  if (siteId === undefined) {
+    const company = requireCompanyId(input.company)
+    const sites = await input.getAllSites(input.apiKey, company, true)
+    const selectedSite = await resolveSite(sites, compact([siteId, site]), site)
```
[the remark this answers is not in the document]
Comment (codepuncher) [reply src/commands/kinsta/env/push.ts:99]: Fixed in 0cebd29. When `--site_id` is not provided and site resolution falls back to company lookup, we now fail fast with `No Kinsta sites found for company "<company>"` before calling `resolveSite()`, matching `env open` behaviour.

File: src/commands/kinsta/env/push.ts [hunk 12]
```diff
@@ -1,26 +1,258 @@
...
+
+  return normalizedCompany
+}
+
+const validateSiteIdAndNameMatch = async (input: ResolvePushTargetIdsInput, siteId: string, site: string | undefined): Promise<void> => {
+  const normalizedCompany = input.company.trim()
+  if (site === undefined || normalizedCompany.length === 0) {
+    return
+  }
+
```
[the remark this answers is not in the document]
Comment (codepuncher) [reply src/commands/kinsta/env/push.ts:71]: Fixed in 40d642a. `--site` with `--site_id` now requires `--company`, so we always validate the two values deterministically instead of silently ignoring `--site` when company context is missing.
